llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -5015,6 +5015,22 @@ Instruction *InstCombinerImpl::foldICmpBinOp(ICmpInst &I,
     }
   }
 
+  // (icmp eq/ne (X, -P2), INT_MIN)
+  //	-> (icmp slt/sge X, INT_MIN + P2)
+  if (ICmpInst::isEquality(Pred) && BO0 &&
+      match(I.getOperand(1), m_SignMask())) {
+    Value *X;
+    if (match(BO0, m_And(m_Value(X), m_CheckedInt([](const APInt &C) {
+                           return C.isZero() || C.isNegatedPowerOf2();
+                         })))) {
```

Comment:
Use `m_NegatedPower2OrZero` instead?

Reply:
Since that matches a splat I cant imagine the zero is necessary.

Reply:
m_NegatedPower2OrZero should also match non-splat (the variant without APInt argument).